apps/native/app/src/screens/air-discount/air-discount.tsx

```diff
@@ -0,0 +1,278 @@
+    error: flightLegsError,
+  } = useGetAirDiscountFlightLegsQuery({
+    fetchPolicy: 'cache-and-network',
+  })
+
+  const flightLegs = flightLegsData?.airDiscountSchemeUserAndRelationsFlights
+
+  const connectionCodes = data?.airDiscountSchemeDiscounts?.filter(
+    (discount) => discount.connectionDiscountCodes.length > 0,
+  )
+
+  const noRights =
+    data?.airDiscountSchemeDiscounts?.filter(
+      (item) => item.user.fund?.credit === 0 && item.user.fund.used === 0,
+    ).length === data?.airDiscountSchemeDiscounts?.length
+
+  return (
+    <ScrollView style={{ flex: 1 }}>
+      <SafeAreaView style={{ marginHorizontal: theme.spacing[2] }}>
+        <Heading>
+          <FormattedMessage
+            id="airDiscount.headingTitle"
+            defaultMessage="Lægra fargjald með loftbrú"
+          />
+        </Heading>
+        <Typography>
+          <FormattedMessage
+            id="airDiscount.headingSubtitle"
+            defaultMessage="Hver einstaklingur með lögheimili innan skilgreinds svæðis á rétt á afslætti á sex flugleggjum á ári. með notkun afsláttar með Loftbrú staðfestir þú að hafa lesið notendaskilmála Loftbrúar."
+          />
+        </Typography>
+        <BulletList>
+          <Bullet>
+            <FormattedMessage
+              id="airDiscount.bulletPointDiscount"
+              defaultMessage="Hver afsláttur nemur 40% af flugfargjaldi."
+            />
+          </Bullet>
+          <Bullet>
+            <FormattedMessage
+              id="airDiscount.bulletPointUsage"
+              defaultMessage="Kóðinn virkjar afslátt í bókunarvél félagsins."
+            />
+          </Bullet>
+        </BulletList>
+        <TOSLink>
+          <Link
+            url={
+              'https://island.is/loftbru/notendaskilmalar-vegagerdarinnar-fyrir-loftbru'
+            }
+          >
+            <LinkText>
+              <FormattedMessage
+                id="airDiscount.tosLinkText"
+                defaultMessage="Notendaskilmálar"
+              />
+            </LinkText>
+          </Link>
+        </TOSLink>
+
+        {(loading || flightLegsLoading) && !error && <SkeletonItem />}
+        {!loading && !error && noRights && <Empty />}
+
+        {data && !noRights && (
+          <View>
+            <AlertWrapper>
+              <Alert
+                type="warning"
+                title={intl.formatMessage({ id: 'airDiscount.alertTitle' })}
+                message={intl.formatMessage({
+                  id: 'airDiscount.alertDescription',
+                })}
+                hasBorder={true}
+              />
+            </AlertWrapper>
+
+            <Typography weight="600" style={{ marginBottom: theme.spacing[1] }}>
+              <FormattedMessage
+                id="airDiscount.myRights"
+                defaultMessage="Mín réttindi"
+              />
+            </Typography>
+            {data?.airDiscountSchemeDiscounts
+              ?.filter(
+                (discount) =>
```

Comment:
~nit: You can extract user from discount for shorter code or better yet to move filter into a separate function